<launch>

  <arg name="use_urdf" default="true"/>
  <arg name="kinova_robotType" default="j2n6s300" />
  <arg name="kinova_robotName" default="$(arg kinova_robotType)"/>
	<arg name="kinova_robotSerial" default="not_set" />

  <!-- If the node handles multiple robots uncomment this and configure /config/multiple_robots.yaml" -->
	<rosparam file="$(find kinova_bringup)/launch/config/multiple_robots.yaml" command="load" />

  <node name="$(arg kinova_robotName)_driver" pkg="kinova_driver" type="kinova_arm_driver" output="screen" cwd="node" args="$(arg kinova_robotType)">
    <rosparam file="$(find kinova_bringup)/launch/config/robot_parameters.yaml" command="load" />
    <param name="serial_number" value="$(arg kinova_robotSerial)" />   
    <param name="robot_name" value="$(arg kinova_robotName)" />   
    <param name="robot_type" value="$(arg kinova_robotType)" />   
  </node>
    
  <group unless="$(arg use_urdf)">
  <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
  <include file="$(find j2n6s300_moveit_config)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>
  </group>

  <group if="$(arg use_urdf)">
    <param name="robot_description" command="$(find xacro)/xacro.py '$(find kinova_description)/urdf/$(arg kinova_robotType)_standalone.xacro'" />
    <node name="$(arg kinova_robotName)_state_publisher"
           pkg="robot_state_publisher"
          type="robot_state_publisher">
      <remap from="joint_states" to="/$(arg kinova_robotName)_driver/out/joint_state"/>
    </node>
  </group>

</launch>

